src/jdk.jdi/share/classes/com/sun/jdi/ThreadReference.java

```diff
@@ -116,18 +116,20 @@ public interface ThreadReference extends ObjectReference {
      * A debugger thread in the target VM will stop this thread
      * with the given {@link java.lang.Throwable} object.
      * <p>
-     * The target VM may not support, or may only provide limited support,
-     * for stopping a virtual thread with an asynchronous exception. It may,
-     * for example, only support this operation when the virtual thread is
-     * suspended at a breakpoint or singlestep event.
+     * This method may be used to send an asynchronous
+     * exception to a virtual thread when it is suspended at an event.
+     * An implementation may support sending an asynchronous exception
+     * to a suspended virtual thread in other cases.
+
      *
      * @param throwable the asynchronous exception to throw
      * @throws InvalidTypeException if <code>throwable</code> is not
      * an instance of java.lang.Throwable in the target VM
-     * @throws IllegalThreadStateException if the thread has terminated
-     * @throws UnsupportedOperationException if the thread is a virtual
-     * thread and the target VM does not support this operation on
-     * virtual threads
+     * @throws IllegalThreadStateException if the thread has terminated,
+     * or if the thread is a virtual thread and was not suspended
+     * @throws OpaqueFrameException if the thread is a suspended
+     * virtual thread and the implementation was unable to throw an
+     * asynchronous exception from the current frame
```

Comment:
I think "from the thread's current frame" might be a bit clearer here, more so than the JDWP spec because the current frame could be read as the current JDI (if you see what I mean). What you have is okay too, but like the JDWP spec, could be a bit clearer.

Reply:
Ok, but JVMTI doesn't use the wording you suggest, and it seems it is in the same situation.

```
        <error id="JVMTI_ERROR_OPAQUE_FRAME">
          The thread is a suspended virtual thread and the implementation was unable
          to throw an asynchronous exception from the current frame.
```

Reply:
> Ok, but JVMTI doesn't use the wording you suggest, and it seems it is in the same situation.

I don't disagree, it's just a bit more pronounced with JDI because the caller is in different VM. It's the "current frame" of the Thread in the target VM rather than the current VM.  I'll leave it to you as to whether you want to re-visit this wording or not.

Reply:
I've updated the text.